### PR TITLE
Deploy MULTI_KIND objects, objects only once

### PR DESCRIPTION
**Pull request by fejta**
/assign @krzyzacy @BenTheElder @cjwagner 

Sen noticed that we didn't deploy anything with MULTI_KIND type. Added an "all" type for to `component()` for this.

Cole noticed that we deploy everything twice, which is because we added both `:hook` and `:deployment` types. Switching to just use the all type, which now instead lists `:hook_deployment, :hook_service`

### CODE REVIEW

File: prow/def.bzl [hunk 1]
```diff
@@ -138,19 +140,21 @@ def component(cmd, *kinds, **kwargs):
...
-#   k8s_objects(name = "fancy", objects=[":hook", ":plank", ":deployment", ":service"])
+#   k8s_objects(name = "staging", objects=[":hook_deployment", ":hook_service", ":plank_deployment"])
 def release(name, *components):
   targets = {}
   objs = []
   for cs in components:
     for (n, ts) in cs.items():
-      targets.setdefault(n, []).extend(ts)
+      if n == "all":
+        objs.extend(ts)
```
Comment (krzyzacy) [inline prow/def.bzl]: just create a `k8s_objects(name=name, objects=ts)` here and you can drop references to `obj`?

**Comment by krzyzacy** [comment]
(ref https://github.com/kubernetes/test-infra/issues/9902)

File: prow/def.bzl (hunk 1 above)
Comment (fejta) [reply prow/def.bzl]: No that won't work how you are suggesting. We would at best wind up with all the targets from a single component or in reality error out with a "redefining target %s" % name as soon as the second component exists.

```
components["hook"] = {"all": ["hook_deployment", "hook_service"], "hook": ["hook_deployment", "hook_service"], "deployment": ["hook_deployment"], "service": ["hook_service"]}
components["deck"] = {"all": ["deck_deployment", "deck_service"], "deck": ["deck_deployment", "deck_service"], "deployment": ["deck_deployment"], "service": ["deck_service"]}
```

`ts` will be either `["deck_deployment", "deck_service"]` or `["hook_deployment", "hook_service"]` 

And we want `k8s_objects(name=name, objs=objs)`, where `objs` is the union of all alls, which is what happens on L158

**Pull request merged**